matrix-client/src/Network/Matrix/Client.hs

```diff
@@ -179,12 +180,16 @@ mkLoginRequest LoginCredentials {..} =
 
 -- | 'login' allows you to generate a session token.
 login :: LoginCredentials -> IO ClientSession
-login cred = do
+login = fmap fst . loginToken 
```

Comment:
It seems like the `login` function is prone to leak access token. I wonder if it should be updated into a `withLogin` using bracket to ensure logout is called. Otherwise we might want to remove it.